hist/histv7/inc/ROOT/RHistAutoAxisFiller.hxx

```diff
@@ -0,0 +1,195 @@
+
+private:
+   void BufferImpl(double x, RWeight weight)
+   {
+      if constexpr (SupportsWeightedFilling) {
+         fBuffer.emplace_back(x, weight);
+      } else {
+         assert(weight.fValue == 1.0);
+         // Silence compiler warning about unused parameter
+         (void)weight;
+         fBuffer.push_back(x);
+      }
+      if (x < fMinimum) {
+         fMinimum = x;
+      }
+      if (x > fMaximum) {
+         fMaximum = x;
+      }
+
+      if (fBuffer.size() >= fMaxBufferSize) {
+         Flush();
+      }
+   }
+
+public:
+   /// Fill an entry into the histogram.
+   ///
+   /// \param[in] x the argument
+   /// \par See also
+   /// the \ref Fill(double x, RWeight weight) "overload for weighted filling"
+   void Fill(double x)
+   {
+      // If the histogram exists, forward the Fill call.
+      if (fHist) {
+         fHist->Fill(x);
+         return;
+      }
+      BufferImpl(x, RWeight(1.0));
+   }
+
+   /// Fill an entry into the histogram with a weight.
+   ///
+   /// This overload is only available for floating-point bin content types (see
+   /// \ref RHistEngine::SupportsWeightedFilling).
+   ///
+   /// \param[in] x the argument
+   /// \param[in] weight the weight for this entry
+   /// \par See also
+   /// the \ref Fill(double x) "overload for unweighted filling"
+   void Fill(double x, RWeight weight)
+   {
+      // If the histogram exists, forward the Fill call.
+      if (fHist) {
+         fHist->Fill(x, weight);
+         return;
+      }
+      BufferImpl(x, weight);
+   }
+
+   /// Flush the buffer of entries and construct the histogram.
+   ///
+   /// Throws an exception if the buffer is empty, the axis interval cannot be determined, or if it would be empty
+   /// because the minimum equals the maximum.
+   void Flush()
+   {
+      if (fHist) {
+         assert(fBuffer.empty() && "buffer should have been emptied");
+         return;
+      }
+
+      if (fBuffer.empty()) {
+         throw std::runtime_error("buffer is empty, cannot create histogram");
+      }
+      if (!std::isfinite(fMinimum) || !std::isfinite(fMaximum)) {
+         throw std::runtime_error("could not determine axis interval");
+      }
+      if (fMinimum == fMaximum) {
+         throw std::runtime_error("axis interval is empty");
+      }
+
+      // Slightly increase the upper limit to make sure the maximum is included in the last bin.
+      double high = std::nextafter(fMaximum, std::numeric_limits<double>::infinity());
+      assert(high > fMaximum);
+      fHist.emplace(fNNormalBins, std::make_pair(fMinimum, high));
+
+      for (auto &&x : fBuffer) {
+         if constexpr (SupportsWeightedFilling) {
+            fHist->Fill(x.first, x.second);
+         } else {
+            fHist->Fill(x);
+         }
+      }
+      fBuffer.clear();
+   }
+
+   /// Return the constructed histogram.
+   ///
+   /// \see Flush()
+   RHist<BinContentType> &GetHist()
+   {
+      Flush();
```

Comment:
That means that we can't get an empty histogram from the auto filler. Is this the desired behavior?